src/OrleansRuntime/Catalog/GrainCreator.cs

```diff
@@ -32,7 +33,7 @@ public GrainCreator(IGrainRuntime grainRuntime, IServiceProvider services = null
         public Grain CreateGrainInstance(Type grainType, IGrainIdentity identity)
         {
             var grain = _services != null
-                ? (Grain) _services.GetService(grainType)
+                ? (Grain) _services.GetRequiredService(grainType)
                 : (Grain) Activator.CreateInstance(grainType);
```

Comment:
It is, since in the StartupBuilder on the last line a DefaultServiceProvider was returned which I considered "bad" behavior, since that part is responsible for locating and loading and invoking the Startup type. This is how it was modified here.

IMHO if someone configures a Startup class and we can't find a delegate which returns a configured ServiceCollection we should return an Exception, since that is bad configuration, but did not want to force my opinion now ;-)